src/Surface.zig

```diff
@@ -232,6 +238,273 @@ pub const Keyboard = struct {
+    /// going to the program running in the terminal.
+    search_active: bool = false,
+
+    /// Initialize a new search manager. This sets up all the internal
+    /// structures needed for searching, but doesn't start any searches.
+    /// Users need to call startSearch() when they want to find something.
+    pub fn init(_: Allocator) SearchManager {
+        return SearchManager{
+            // All fields start with their default values defined above
+        };
+    }
+
+    /// Clean up all memory used by the search manager.
+    /// This cancels any running searches and frees all result data.
+    /// Call this when the terminal window is being closed.
+    pub fn deinit(self: *SearchManager, alloc: Allocator) void {
+        // Cancel any search that's currently running
+        self.cancelCurrentSearch(alloc);
+
+        // Free the search text if we have any
+        if (self.search_needle) |needle| {
+            alloc.free(needle);
+            self.search_needle = null;
+        }
+
+        // Free all the search results
+        self.results.deinit(alloc);
+
+        // Clean up thread pool if we created one
+        if (self.thread_pool) |pool| {
+            pool.deinit();
+            alloc.destroy(pool);
+            self.thread_pool = null;
+        }
+    }
+
+    /// Cancel the search that's currently running, if any.
+    /// This stops any background work and cleans up resources.
+    /// It's safe to call this even if no search is active.
+    fn cancelCurrentSearch(self: *SearchManager, alloc: Allocator) void {
+        if (self.current_search) |search| {
+            search.deinit(alloc);
+            alloc.destroy(search);
+            self.current_search = null;
+        }
+        self.search_in_progress = false;
+    }
+
+    /// Start searching for the given text in terminal history.
+    /// This function returns immediately and does the real work in a
+    /// background thread to keep the user interface smooth.
+    /// Results are delivered through a callback when the search completes.
+    pub fn startSearch(self: *SearchManager, alloc: Allocator, needle: []const u8, page_list: *terminal.PageList) !void {
+        // Cancel any search that's already running
+        self.cancelCurrentSearch(alloc);
+
+        // Clear old results
+        self.results.clearAndFree(alloc);
+        self.current_result_idx = 0;
+
+        // Save the new search text
+        if (self.search_needle) |old_needle| {
+            alloc.free(old_needle);
+        }
+        self.search_needle = try alloc.dupe(u8, needle);
+
+        // Mark that we're starting a search
+        self.search_in_progress = true;
+
+        // Create or get thread pool for background search work
+        if (self.thread_pool == null) {
+            self.thread_pool = try alloc.create(std.Thread.Pool);
+            self.thread_pool.?.* = std.Thread.Pool{};
+            try self.thread_pool.?.init(.{ .allocator = alloc, .n_jobs = 1 });
```

Comment:
Style (avoid `.?` if you can, avoid explicit types)
```suggestion
            const pool = try alloc.create(std.Thread.Pool);
            pool.* = .{};
            try pool.init(.{ .allocator = alloc, .n_jobs = 1 });
            self.thread_pool = pool;
```